app/lib/backend/model_extensions/metric.rb

```diff
@@ -5,8 +5,7 @@ def self.included(base)
         base.class_eval do
           before_destroy :cache_service_association
 
-          # WARN: last callback is called first, gotcha!
-          after_commit :sync_backend, if: ->(m) { (m.persisted? || m.destroyed?) && !m.changed? }
+          after_commit :sync_backend, on: %i[create destroy]
```

Comment:
Wow! 😱 It is failing here... I wasn't expecting this one 😞  
```ruby
test 'sync backend metric data when metric is updated' do
  service = FactoryBot.create(:simple_service)
  metric = Metric.new(service: service, system_name: 'foos', friendly_name: 'Foos', unit: 'foo')

  BackendMetricWorker.expects(:sync)
  metric.save!

  metric.system_name = 'bars'
  BackendMetricWorker.expects(:sync).with(service.backend_id, metric.id, metric.name)

  metric.save!
end
```

I will investigate a bit about this 🤔 👍